src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1166,6 +1166,53 @@ internal virtual void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleB
         /// </summary>
         public abstract bool AreLocalsZeroed { get; }
 
+        // PROTOTYPE: Cache the result.
+        internal virtual bool? IsNullableEnabled()
+        {
+            bool? result = false;
+            if (this is SourceMemberMethodSymbol sourceMethod)
+            {
+                result = isNullableEnabled(sourceMethod.SyntaxNode);
```

Comment:
It looks like this property returns ```CSharpSyntaxNode```. How do we know that this is the right node to look at for all existing flavors of symbols and will work for symbols we might add in the future. Looking at this code, I am unable to reason what we are looking at.